### PR TITLE
Disable integrations when Central is managed

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

This disables integrations when Central is managed. This covers is from multiple perspectives. Central-side and Sensor side

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Will see if the log line is ever triggered and check that autogenerated registries do not show up

### CODE REVIEW

**Review by dcaravel — APPROVED** [review]
/LGTM

File: sensor/common/managedcentral/managed.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	managedCentral bool
+	managedMutex   sync.RWMutex
+)
+
+// IsCentralManaged returns if Central is managed
+func IsCentralManaged() bool {
```
Comment (RTann) [inline sensor/common/managedcentral/managed.go:13]: I'm a bit confused. Is this not already known from the environment variable?

File: sensor/common/managedcentral/managed.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	managedMutex.RLock()
+	defer managedMutex.RUnlock()
+
+	return managedCentral
+}
+
+// Set sets the value of whether Central is managed
+func Set(value bool) {
```
Comment (RTann) [inline sensor/common/managedcentral/managed.go:21]: Is this not only set by a single goroutine?

File: sensor/common/managedcentral/managed.go (hunk 1 above)
Comment (RTann) [reply sensor/common/managedcentral/managed.go:13]: oh is that only set in Central? I feel like Central can set this when it makes related Sensor bundles, right?

Comment (connorgorman) [reply sensor/common/managedcentral/managed.go:13]: So we can, but there are older sensors that could be connected that will not have it. Also existing sensors that are upgraded will not have it

**Pull request merged**